docs/DeveloperGuide.md

```diff
@@ -4,9 +4,457 @@
 
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
-## Design & implementation
+ical4J Library: https://www.ical4j.org/
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+
+
+## Architecture
+The Calendar application is designed with a modular architecture, consisting of the following components:
+
+
+#### Main Components
+- **UI**: The `UiRenderer` class handles the rendering of the calendar views (week and month) to the console.
+- **Logic**: The `Main` class acts as the central logic component, handling user input and dispatching commands to the appropriate components.
+- **Data**: The `TaskManager` class manages the tasks and their corresponding dates, providing methods for adding, updating, and deleting tasks.
+- **Storage**: The `Storage` class handles the persistence of tasks by reading from and writing to a file.
+- **Time**: The `View` class and its subclasses (`WeekView` and `MonthView`) manage the rendering and navigation of the week and month views.
+
+
+## Data Component
+### API: [Data](https://github.com/AY2324S2-CS2113-W13-2/tp/tree/master/src/main/java/data)
+![Data Class Diagram](images/class/Data.png)
```

Comment:
Could the visual be split up into smaller parts?
![image](https://github.com/nus-cs2113-AY2324S2/tp/assets/96415327/1f35598c-e11a-47de-b417-41f28df6b11b)